src/SprykerEco/Zed/Payone/Business/Payment/PaymentDetail.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Payone\Business\Payment;
+
+use Generated\Shared\Transfer\PaymentDetailTransfer;
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class PaymentDetail implements PaymentDetailInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     */
+    public function __construct(PayoneQueryContainerInterface $queryContainer)
+    {
+        $this->queryContainer = $queryContainer;
+    }
+
+    /**
+     * @param int $idOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentDetailTransfer
+     */
+    public function getPaymentDetail(int $idOrder): PaymentDetailTransfer
+    {
+        $paymentEntity = $this->queryContainer->createPaymentByOrderId($idOrder)->findOne();
+        $paymentDetailEntity = $paymentEntity->getSpyPaymentPayoneDetail();
+        $paymentDetailTransfer = new PaymentDetailTransfer();
+        $paymentDetailTransfer->fromArray($paymentDetailEntity->toArray(), true);
+
+        return $paymentDetailTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentDetailTransfer $paymentDataTransfer
+     * @param int $idOrder
+     *
+     * @return void
+     */
+    public function updatePaymentDetail(PaymentDetailTransfer $paymentDataTransfer, $idOrder): void
```

Comment:
* Move this method to a `\SprykerEco\Zed\Payone\Persistence\PayoneRepository`.

Reply:
Need discussion

Reply:
++